src/librbd/CopyupRequest.cc

```diff
@@ -181,6 +181,19 @@ bool CopyupRequest::send_copyup() {
   return false;
 }
 
+bool CopyupRequest::is_nop() {
+  bool noop = true;
+  for (const AioObjectRequest<> *req : m_pending_requests) {
+    const AioObjectWrite *wreq = dynamic_cast<const AioObjectWrite *>(req);
```

Comment:
I'd eliminate the dynamic_cast and just add a new pure virtual method to the base object request class that returns true/false for whether or not it's an empty write op